rest/config_legacy.go

```diff
@@ -93,17 +97,28 @@ type UnsupportedServerConfigLegacy struct {
 // ToStartupConfig returns the given LegacyServerConfig as a StartupConfig and a set of DBConfigs.
 // The returned configs do not contain any default values - only a direct mapping of legacy config options as they were given.
 func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, error) {
-
 	// find a database's credentials for bootstrap (this isn't the first database config entry due to map iteration)
 	bsc := &BootstrapConfig{}
 	for _, dbConfig := range lc.Databases {
 		if dbConfig.Server == nil || *dbConfig.Server == "" {
 			continue
 		}
+
+		server, username, password, err := gocbv1tov2AddressUpgrade(*dbConfig.Server)
+		if err != nil {
+			return nil, DbConfigMap{}, err
+		}
+
+		// Prioritise config fields over credentials in host
+		if dbConfig.Username != "" && dbConfig.Password != "" {
+			username = dbConfig.Username
+			password = dbConfig.Password
+		}
```

Comment:
I think we want to do this instead, and only fall back to connstr creds when both are blank.

```suggestion
		// Prioritise config fields over credentials in host
		if dbConfig.Username != "" || dbConfig.Password != "" {
			username = dbConfig.Username
			password = dbConfig.Password
		}
```